backend/src/main/java/org/example/app/task/logic/UcSaveTaskItem.java

```diff
@@ -21,16 +22,22 @@ public class UcSaveTaskItem {
   @Inject
   TaskItemMapper taskItemMapper;
 
+  @Inject
+  TaskListRepository taskListRepository;
+
   /**
    * @param item the {@link TaskItemEto} to save.
    * @return the {@link TaskItemEntity#getId() primary key} of the saved {@link TaskItemEntity}.
    */
   // @RolesAllowed(ApplicationAccessControlConfig.PERMISSION_SAVE_TASK_ITEM)
   public Long save(TaskItemEto item) {
 
-    TaskItemEntity entity = this.taskItemMapper.toEntity(item);
-    entity = this.taskItemRepository.save(entity);
-    return entity.getId();
+    final TaskItemEntity entity = this.taskItemMapper.toEntity(item);
+    if (item.getTaskListId() != null) {
+      taskListRepository.findById(item.getTaskListId()).ifPresent(entity::setTaskList);
+    }
+    TaskItemEntity savedEntity = this.taskItemRepository.save(entity);
+    return savedEntity.getId();
```

Comment:
This looks more like a hack than a solution to me.
So if I get it right the relation from item to list got lost here.
That means that hibernate (new version) or its quarkus implementation behaves different to what we have done before with devon4j and spring.
The other "hack" I am talking about is this one:
https://github.com/devonfw-training/java-quarkus/blob/09459902774bec949586ef124e06639e9cfd69d1/backend/src/main/java/org/example/app/task/dataaccess/TaskItemEntity.java#L70

There was also a relatively clean variant in devon4j for this but it was not properly used and accepted:
https://github.com/devonfw-training/java-spring/blob/solution/demo/core/src/main/java/org/example/demo/ordermanagement/dataaccess/api/ItemEntity.java#L91
https://github.com/devonfw/devon4j/blob/b6e54ab5df028bd17a15a76ee72ac7dfeb179a48/modules/jpa-basic/src/main/java/com/devonfw/module/jpa/dataaccess/api/JpaHelper.java#L44